### PR TITLE
2870 additional logging around login session creation

### PR DESCRIPTION
**Pull request by cadibemma**
## Description of change
To have more insight – and be able to respond more quickly – on the sign in > introspect > profile creation funnel inside vets-api. 

## Original issue(s)
department-of-veterans-affairs/va-mobile-app#2870

### CODE REVIEW

**Review by kpethtel — COMMENTED** [review]
I have a few thoughts. LMK if you want to talk about this via slack or huddle.

File: lib/iam_ssoe_oauth/session_manager.rb [hunk 1]
```diff
@@ -62,6 +66,15 @@ def create_user_session
       raise e
     end
 
+    def call_to_introspect_check(iam_profile)
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply lib/iam_ssoe_oauth/session_manager.rb]: i believe i fixed this

File: config/initializers/statsd.rb [hunk 2]
```diff
@@ -323,9 +323,13 @@
                                                     'iam_ssoe_oauth.create_user_session'
   IAMSSOeOAuth::SessionManager.statsd_measure :create_user_session,
                                               'iam_ssoe_oauth.create_user_session.measure'
+
   StatsD.increment('iam_ssoe_oauth.create_user_session.success', 0)
   StatsD.increment('iam_ssoe_oauth.create_user_session.failure', 0)
   StatsD.increment('iam_ssoe_oauth.inactive_session', 0)
+  StatsD.increment('iam_ssoe_oauth.user_sign_in.success', 0)
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply config/initializers/statsd.rb]: As discussed, its to have checkpoints to infer where things may be failing within `create_user_session`

File: lib/iam_ssoe_oauth/session_manager.rb [hunk 3]
```diff
@@ -84,6 +97,8 @@ def build_identity(iam_profile)
 
     def build_session(access_token, user_identity)
       @session = IAMSession.new(token: access_token, uuid: user_identity.uuid)
+      Rails.logger.info('IAMUser create user session: build session started', @session.uuid)
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply lib/iam_ssoe_oauth/session_manager.rb]: you are right. Its been changed

File: lib/iam_ssoe_oauth/session_manager.rb [hunk 4]
```diff
@@ -44,24 +44,39 @@ def logout
...
     rescue Common::Exceptions::Unauthorized => e
       Rails.logger.error('IAMUser create user session: unauthorized', error: e.message)
       StatsD.increment('iam_ssoe_oauth.inactive_session')
       raise e
     end
 
+    def build_profile(access_token)
+      iam_profile = iam_ssoe_service.post_introspect(access_token)
+
+      if iam_profile
```
Comment (kpethtel) [inline lib/iam_ssoe_oauth/session_manager.rb:72]: As we discussed before, this guard doesn't do anything.

File: lib/iam_ssoe_oauth/session_manager.rb [hunk 5]
```diff
@@ -44,24 +44,39 @@ def logout
...
+      user = persist(session, user)
+      StatsD.increment('iam_ssoe_oauth.user_session_creation_done')
+      user
     rescue Common::Exceptions::Unauthorized => e
       Rails.logger.error('IAMUser create user session: unauthorized', error: e.message)
       StatsD.increment('iam_ssoe_oauth.inactive_session')
       raise e
     end
 
+    def build_profile(access_token)
```
[the remark this answers is not in the document]
Comment (cadibemma) [reply lib/iam_ssoe_oauth/session_manager.rb]: fixed

File: lib/iam_ssoe_oauth/session_manager.rb (hunk 4 above)
Comment (cadibemma) [reply lib/iam_ssoe_oauth/session_manager.rb]: its now removed

**Review by kpethtel — APPROVED** [review]
👍

**Review by rileyanderson — APPROVED** [review]

**Pull request merged**